core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,372 @@
+    throw new UnsupportedOperationException(
+        "Cannot create TableScan from table of type POSITION_DELETES");
+  }
+
+  @Override
+  public BatchScan newBatchScan() {
+    return new PositionDeletesTableScan(operations(), table(), schema());
+  }
+
+  @Override
+  public Schema schema() {
+    return schema;
+  }
+
+  private Schema calculateSchema() {
+    Types.StructType partitionType = Partitioning.partitionType(table());
+    Schema result =
+        new Schema(
+            MetadataColumns.DELETE_FILE_PATH,
+            MetadataColumns.DELETE_FILE_POS,
+            Types.NestedField.optional(
+                MetadataColumns.DELETE_FILE_ROW_FIELD_ID,
+                "row",
+                table().schema().asStruct(),
+                MetadataColumns.DELETE_FILE_ROW_DOC),
+            Types.NestedField.required(
+                POSITION_DELETE_TABLE_PARTITION_FIELD_ID,
+                "partition",
+                partitionType,
+                "Partition that position delete row belongs to"),
+            Types.NestedField.required(
+                POSITION_DELETE_TABLE_SPEC_ID,
+                "spec_id",
+                Types.IntegerType.get(),
+                "Spec ID of the file that the position delete row belongs to"),
+            Types.NestedField.required(
+                POSITION_DELETE_TABLE_FILE_PATH,
+                "delete_file_path",
+                Types.StringType.get(),
+                "Spec ID of the file that the position delete row belongs to"));
+
+    if (partitionType.fields().size() > 0) {
+      return result;
+    } else {
+      // avoid returning an empty struct, which is not always supported. instead, drop the partition
+      // field
+      return TypeUtil.selectNot(result, Sets.newHashSet(POSITION_DELETE_TABLE_PARTITION_FIELD_ID));
+    }
+  }
+
+  public static class PositionDeletesTableScan
+      extends SnapshotScan<BatchScan, ScanTask, ScanTaskGroup<ScanTask>> implements BatchScan {
+
+    protected PositionDeletesTableScan(TableOperations ops, Table table, Schema schema) {
+      super(ops, table, schema, new TableScanContext());
+    }
+
+    protected PositionDeletesTableScan(
+        TableOperations ops, Table table, Schema schema, TableScanContext context) {
+      super(ops, table, schema, context);
+    }
+
+    @Override
+    protected PositionDeletesTableScan newRefinedScan(
+        TableOperations newOps, Table newTable, Schema newSchema, TableScanContext newContext) {
+      return new PositionDeletesTableScan(newOps, newTable, newSchema, newContext);
+    }
+
+    @Override
+    public CloseableIterable<ScanTaskGroup<ScanTask>> planTasks() {
+      CloseableIterable<ScanTask> scanTasks = planFiles();
+      return TableScanUtil.planTaskGroups(
+          scanTasks, targetSplitSize(), splitLookback(), splitOpenFileCost());
+    }
+
+    @Override
+    protected CloseableIterable<ScanTask> doPlanFiles() {
+      Expression rowFilter = context().rowFilter();
```

Comment:
nit: We could just use `filter()` directly, which wraps `context().rowFilter()`.

Reply:
Fixed